bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -236,21 +236,57 @@ public void controlResized(ControlEvent e) {
 	private IPartListener partListener = new IPartListener() {
 		@Override
 		public void partActivated(IWorkbenchPart part) {
-			if (getShell() != null) {
-				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
-			}
+			Display.getDefault().asyncExec(new Runnable() {
```

Comment:
It would be much easier to read with `this::reactToListener`. Same for the code below.

Reply:
@iloveeclipse I don't quite get what you want: do you want me to move the method out of the Listener class?

Reply:
No, I just wanted to say that inner classes for single accessible methods could be replaced with method references (5 lines vs one line).

Reply:
I got you! Fixed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/a5a91226dc78be4b66fe690aae04ab5c19d58c8e